✨ UI Changes - Removing usage of PersonalSidebar

Sean-Stilwell opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->

## Description
<!-- Please provide a brief description of the changes made in this pull request -->

This PR makes a series of UI changes requested by Deb. It builds off the changes made in #1282:

- The PersonalSidebar is no longer displayed.
    - Workspace and Learn sidebars are still displayed.
- A new Settings page is added that combines everything previously under /account/.
- Cards for Recent Links, Create a New Workspace, and to access the Explore page are added to the Home page.
- My Workspaces is now split into Admin/Lead, Collaborator, and Guest workspaces (if a user has any for each role).
- Minor changes to naming of pages.

![image](https://github.com/user-attachments/assets/0e61aec9-2480-4b1e-a4c1-20ee0ceae10e)

![image](https://github.com/user-attachments/assets/69def20a-c1da-40b9-8d70-b81515a2c2d5)

![image](https://github.com/user-attachments/assets/cfb1a18a-ff88-4f7b-b7c4-bb3f2a928cb2)

## How Has This Been Tested?
<!-- Please describe how you tested your changes. -->
<!-- Have you created new tests or updated existing ones? -->
<!-- e.g. unit | visually | none -->

Locally tested

## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Unit test (new or update to tests)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (change that would cause documentation to be updated)
- [ ] Configuration change (change that would affect the configuration of the application)

<!-- If you made any visual changes, provide screenshots of before/after, it its moving parts, please provide high quality gif, wemb or mp4 -->

## Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask! -->
- [x] PR is submitted to the correct branch `develop`.
- [x] Code follows the code style of this project.
- [x] Changes are covered by Specflow tests and all new or existing tests are passing.
- [x] Any new or updated translatable strings have been added to the localization files.
- [x] Necessary and relevant documentation has been created or updated.
- [x] There are either no changes to the application's configuration or the necessary changes in the infrastructure repository are included in a separate PR.

<!-- Link to the "infrastructure" repository PR here -->

Sean-Stilwell committed:
Changing Support Centre to match header

Sean-Stilwell committed:
Placing resources and requests in adjacent cards

Sean-Stilwell committed:
Updating page routes and resource title

Sean-Stilwell committed:
Updating DHButtons to allow lowercase buttons

Sean-Stilwell committed:
Correcting a button label in the topbar

Sean-Stilwell committed:
Swapping the Terms button in footer to DHButton

Sean-Stilwell committed:
Changing title on learn page frequent docs sidebar

Sean-Stilwell committed:
Changing to ul and li elements

Sean-Stilwell committed:
Correcting news label on small screens

Sean-Stilwell committed:
Reverting DHButton, manually setting topbar and footer to use MudText for lowercase

Sean-Stilwell committed:
Adding padding to main page due to appbar

Sean-Stilwell committed:
Splitting user projects list based on role

Sean-Stilwell committed:
Swapping the loading indicator for announcements

Sean-Stilwell committed:
Updating titles for the workspace subsections

Sean-Stilwell committed:
Renaming News to Announcements everywhere

Sean-Stilwell committed:
Reverting underlining the buttons

Sean-Stilwell committed:
Reverting underlining the buttons 2

Sean-Stilwell committed:
Adding spacing to projects component, cleaning it up a bit

Sean-Stilwell committed:
Sidebar only appears on resource and workspace page

Sean-Stilwell committed:
Fixing some weird formatting on the resources page

Sean-Stilwell committed:
Creating settings page with all combined user settings

Sean-Stilwell committed:
Making language toggle behave on new settings page

Sean-Stilwell committed:
Merging changes from first UI changes

Sean-Stilwell committed:
Adding admin toggle and options to Settings page

Sean-Stilwell committed:
Moving some nav options to the home page

Sean-Stilwell committed:
Putting everything into groups for David

Sean-Stilwell committed:
Some tweaks before creating the PR

Sean-Stilwell committed:
Adding localization and ensuring title does not wrap

Sean-Stilwell committed:
Removing references to PersonalSidebar in most locations

Sean-Stilwell commented on Portal/src/Datahub.Portal/Layout/PortalLayout.razor:
Wondering if anyone has a better idea for how to hide the sidebar everywhere except learn/workspaces. This works but not sure if anyone has something cleaner to suggest.

yjmrobert replied:
I wonder if it would make more sense to refactor the `MudDrawer` into a shared component and then we could move it down into the `SectionOutlet`'s own rendering logic

yjmrobert replied:
ie, the pages that want a drawer
```razor
<SectionContent SectionName="side-bar">
  <DHSideDrawer>
    <PersonalSidebar/>
  </DHSideDrawer>
</SectionContent>
```

and then we'd have `DHSideDrawer`
```razor
<MudDrawer
    Width="320px"
    Variant="DrawerVariant.Persistent"
    Open
    Overlay="false"
    Style="@_sideBorderStyle"
    Elevation="0"
    ClipMode="DrawerClipMode.Always" Breakpoint="Breakpoint.Always">
      <MudDrawerContainer>
          @ChildContent
      </MudDrawerContainer>
</MudDrawer>
```

Sean-Stilwell committed:
Merge branch 'develop' into ss/deb-ui-changes-2

yjmrobert reviewed (commented):
Code looks good and lines up with the requested layout - nice work.

I'm still not convinced on the layouts for the landing and settings pages.  I think it merits some further discussions and measuring of where it improves or degrades the user experience before we wholly commit to the changes.

yjmrobert reviewed (changes requested):
Looks like the pictures aren't working on the update
![image](https://github.com/user-attachments/assets/a318f591-0daf-4bc8-8fee-72bae91bd938)

vs 

![image](https://github.com/user-attachments/assets/da31ef8d-4e0a-40ae-86b2-fdc26d81f3d5)


Other than that I'm good to merge it.

yjmrobert committed:
🔧 Refactored side drawer content into a reusable component to simplify rendering logic

Sean-Stilwell committed:
Fixing profile images not appearing

Sean-Stilwell commented:
@yjmrobert Fixed with newest commit

![image](https://github.com/user-attachments/assets/f109c055-87c1-4e15-acbd-4703f8707a15)

yjmrobert committed:
🤫 Reintroduced the sidebar on the accounts page and gave the Home page its own url

yjmrobert committed:
Merge remote-tracking branch 'origin/ss/deb-ui-changes-2' into ss/deb-ui-changes-2

# Conflicts:
#	Portal/src/Datahub.Portal/Pages/Account/PublicProfile/AccountPublicProfileSettings.razor

yjmrobert committed:
🔧 Add Customization section and AccountPublicProfile

Introduced a new Customization section in PersonalSidebar and linked it to AccountPublicProfileSettings. Added AccountPublicProfile component to handle user's public profile information display.
